source/features/stop-pjax-loading-with-esc.tsx

```diff
@@ -0,0 +1,66 @@
+import onetime from 'onetime';
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+const progressLoaderLoadingClass = 'is-loading';
+
+function fixProfileNavAndTimeline() {
+	for (const stickyElement of select.all('.js-sticky:not(.is-stuck)')) {
+		stickyElement.removeAttribute('style');
+	}
+}
+
+function getPjaxContainerId(): string | boolean {
+	const pjaxContainer = select('#js-repo-pjax-container, #js-pjax-container, #gist-pjax-container');
+
+	if (pjaxContainer) {
+		return `#${pjaxContainer.id}`;
+	}
+
+	features.error(__filebasename, 'Pjax container id not found.');
+	return false;
+}
+
+function init() {
+	const progressLoader = select('.progress-pjax-loader')!;
+
+	window.addEventListener('keydown', event => {
+		if (event.key === 'Escape' && progressLoader.classList.contains(progressLoaderLoadingClass)) {
+			const pjaxContainerId = getPjaxContainerId();
+
+			if (pjaxContainerId && history.state && '_id' in history.state) {
```

Comment:
It seems that `pjaxContainerId` is only needed in some cases, but `features.error` logs an error unconditionally. Maybe this should be
```suggestion
			if (history.state && '_id' in history.state) {
				const pjaxContainer = select('#js-repo-pjax-container, #js-pjax-container, #gist-pjax-container');
				if (pjaxContainer) {
					history.replaceState(.........)
				} else {
					features.error(__filebasename, 'Pjax container id not found.');
				}
			}
```